include/ui/hexedit.h

```diff
@@ -71,6 +72,7 @@ class HexEdit : public QAbstractScrollArea {
   void applyChanges();
   void undo();
   void discardChanges();
+  void setUnprintablesMode(QAction* action);
```

Comment:
This is slot for QMenu signal triggered of type void(QAction*) . AFAIK these must have the same type.

Enum instead of string: I'd have to bind somehow action in menu with enum value. Is there any well seen method?

Reply:
You can connect to a lambda which translates the arguments to ones that make more sense. Regarding enum: you can bind each of these actions to a lambda which calls setUnprintablesMode with a proper argument.

Reply:
Done. Not sure if capturing this by value is ok.

Reply:
It's ok, because it's a pointer to an object which will live longer than the lambda.